alica_engine/src/engine/syncmodule/SyncModule.cpp

```diff
@@ -83,8 +83,9 @@ void SyncModule::tick()
 
     // delete the failed processes from the mapping
     for (const SynchronisationProcess* s : endedSyncProcesses) {
-        delete _synchProcessMapping[s->getSynchronisation()];
-        _synchProcessMapping.erase(s->getSynchronisation());
+        const auto* const sync = s->getSynchronisation();
+        delete _synchProcessMapping[sync];
```

Comment:
I'd recommend using an iterator to prevent the repeated map lookup.

Reply:
changed